### PR TITLE
Fix Namespace selector & ensure correct webhook version

### PR DESCRIPTION
**Pull request by jsturtevant**
The name space selectors must be in a specific format https://github.com/kubernetes/kubernetes/issues/92157#issuecomment-825160958. This was causing the webhook to not run validation.

Once those are turned on there is a bug in the upgrade e2e test that leave a webhook around becuase the version of the webhook label is not upgraded and there for not removed by our tooling.

This was found during #5182

Signed-off-by: James Sturtevant <jstur@microsoft.com>

**Description**:

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [x] |
| CLI Tool                   | [x] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [x] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
no

2. Is this a breaking change?
no 

4. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?
no

### CODE REVIEW

**Review by keithmattix — APPROVED** [review]

**Comment by jsturtevant** [comment]
hmm go test failure looks unrelated.

**Review by jaellio — APPROVED** [review]
I think the test failure is related to https://github.com/openservicemesh/osm/pull/5192

**Comment by steeling** [comment]
We should probably add this `createOrUpdateMutatingWebhook` as well

**Comment by jsturtevant** [comment]
> `createOrUpdateMutatingWebhook`

you are right, I meant to do this....

File: pkg/injector/webhook.go [hunk 1]
```diff
@@ -344,7 +344,12 @@ func createOrUpdateMutatingWebhook(clientSet kubernetes.Interface, cert *certifi
 				return err
 			}
 
-			mwhc.ObjectMeta = existing.ObjectMeta // copy the object meta which includes resource version, required for updates.
+			// copy the object meta which includes resource version, required for updates.
+			// but use the OSM Version, which might be different if this is an upgrade
```
Comment (shashankram) [inline pkg/injector/webhook.go:348]: It would be nice to have an example in the comments to make the comment clearer.

Comment (jsturtevant) [reply pkg/injector/webhook.go:348]: something like ?
```suggestion
			// copy the object meta which includes resource version, required for updates.
			// but use the OSM Version, which might be different if this is an upgrade.  For example, 
			// the webhook might have been initially created by version 1.1.2 and the version osm is 1.2.2
```

Comment (shashankram) [reply pkg/injector/webhook.go:348]: It was a bit hard for me to understand the following:
`required for updates`: what updates? (This is an existing comment prior to your change, but seems ambiguous)
`but use the OSM Version which might be different if this is an upgrade`: It's not clear what upgrade this is referring to - k8s, control plane, etc.

Just reading the comments, I found it hard to make sense of the relation between object meta and OSM version, in this conext.

Comment (jsturtevant) [reply pkg/injector/webhook.go:348]: oh yes that makes sense.  Let me see if I can come up with something clearer

Comment (shashankram) [reply pkg/injector/webhook.go:348]: Thanks. This comment isn't blocking, so let me know if you want to merge this as is.

**Pull request merged**